src/main/java/Task.java

```diff
@@ -0,0 +1,120 @@
+import java.util.ArrayList;
+
+public class Task {
+    private String name;
+    private boolean isCompleted;
+
+    public Task(String name, boolean isCompleted) {
+        this.name = name;
+        this.isCompleted = isCompleted;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public boolean getStatus() {
+        return isCompleted;
+    }
+
+    public void setStatus(boolean isCompleted) {
+        this.isCompleted = isCompleted;
+    }
+
+    public static void checkTaskType(String input, ArrayList<Task> listOfTasks) throws InvalidTaskTypeException {
+        String taskType = input;
+        int endTaskTypeIndex = input.indexOf(" ");
+        if (endTaskTypeIndex > 0) {
+            taskType = input.substring(0, endTaskTypeIndex);
+        }
+        switch (taskType) {
+        case "todo":
+            addTodoToTasksList(input, listOfTasks);
+            break;
+        case "deadline":
+            addDeadlineToTasksList(input, listOfTasks);
+            break;
+        case "event":
+            addEventToTasksList(input, listOfTasks);
+            break;
+        default:
+            throw new InvalidTaskTypeException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+        }
+    }
+
+    public static void addTodoToTasksList(String input, ArrayList<Task> listOfTasks) {
+        try {
+            DukeExceptionTest.checkMissingTodoName(input);
+        } catch (MissingTaskNameException e) {
+            System.out.println(e);
+            return;
+        }
+        int nameStartIndex = input.indexOf(" ");
+        String name = input.substring(nameStartIndex + 1);
+        Task todo = new Todo(name, false);
+        listOfTasks.add(todo);
+        printAddTaskMessage(todo, listOfTasks);
+    }
+
+    public static void addDeadlineToTasksList(String input, ArrayList<Task> listOfTasks) {
+        //check min length
+        int nameStartIndex = input.indexOf(" ");
+        int nameEndIndex = input.indexOf("/by");
+        String name = input.substring(nameStartIndex + 1, nameEndIndex -1);
+        //check name length
+        String dueDate = input.substring(nameEndIndex + 4);
```

Comment:
I noticed that you used "magic numbers" like 4 (in this line) and 6 (in line 82). Perhaps you could try processing the inputs in a different way? Personally, I found the String.split() function to be very useful for doing so.